src/main/python/tests/examples/tutorials/test_adult.py

```diff
@@ -0,0 +1,324 @@
+
+        #X = PREPROCESS_package.get_X(X1, 1, train_count)
+        #Y = PREPROCESS_package.get_Y(X1, 1, train_count)
+        #Xt = PREPROCESS_package.get_X(X2, 1, test_count)
+        #Yt = PREPROCESS_package.get_Y(X2, 1, test_count)
+
+        # TODO somehow throws error at predict with this included
+        #X, mean, sigma = scale(X, True, True)
+        #Xt = scaleApply(Xt, mean, sigma)
+
+        betas = multiLogReg(X, Y)
+
+        [_, y_pred, acc] = multiLogRegPredict(Xt, betas, Yt)
+
+        confusion_matrix_abs, _ = confusionMatrix(y_pred, Yt).compute()
+        print(confusion_matrix_abs)
+        self.assertTrue(
+            np.allclose(
+                confusion_matrix_abs,
+                np.array([[11593.,  1545.],
+                          [842., 2302.]])
+            )
+        )
+
+    def test_level3(self):
+        train_count = 32561
+        test_count = 16281
+
+        SCHEMA = '"DOUBLE,STRING,DOUBLE,STRING,DOUBLE,STRING,STRING,STRING,STRING,STRING,DOUBLE,DOUBLE,DOUBLE,STRING,STRING"'
+
+        F1 = self.sds.read(
+            self.dataset_path_train,
+            schema=SCHEMA
+        )
+        F2 = self.sds.read(
+            self.dataset_path_test,
+            schema=SCHEMA
+        )
+
+        jspec = self.sds.read(self.dataset_jspec, data_type="scalar", value_type="string")
+        PREPROCESS_package = self.sds.source(self.preprocess_src_path, "preprocess", print_imported_methods=True)
+
+        X1 = F1.rbind(F2)
+        X1, M1 = X1.transform_encode(spec=jspec)
+
+        X = PREPROCESS_package.get_X(X1, 1, train_count)
+        Y = PREPROCESS_package.get_Y(X1, 1, train_count)
+
+        Xt = PREPROCESS_package.get_X(X1, train_count, train_count + test_count)
+        Yt = PREPROCESS_package.get_Y(X1, train_count, train_count + test_count)
+
+        Yt = PREPROCESS_package.replace_value(Yt, 3.0, 1.0)
+        Yt = PREPROCESS_package.replace_value(Yt, 4.0, 2.0)
+
+        # better alternative for encoding
+        # F2 = F2.replace("<=50K.", "<=50K")
+        # F2 = F2.replace(">50K.", ">50K")
+        # X1, M = F1.transform_encode(spec=jspec)
+        # X2 = F2.transform_apply(spec=jspec, meta=M)
+
+        # X = PREPROCESS_package.get_X(X1, 1, train_count)
+        # Y = PREPROCESS_package.get_Y(X1, 1, train_count)
+        # Xt = PREPROCESS_package.get_X(X2, 1, test_count)
+        # Yt = PREPROCESS_package.get_Y(X2, 1, test_count)
+
+        # TODO somehow throws error at predict with this included
+        # X, mean, sigma = scale(X, True, True)
+        # Xt = scaleApply(Xt, mean, sigma)
+
+        FFN_package = self.sds.source(self.neural_net_src_path, "fnn", print_imported_methods=True)
+
+        epochs = 1
+        batch_size = 16
+        learning_rate = 0.01
+        seed = 42
+
+        network = FFN_package.train(X, Y, epochs, batch_size, learning_rate, seed)
+
+        """
+        If more ressources are available, one can also choose to train the model using a parameter server.
```

Comment:
in general it is actually a bit faster with the parameter server.
the use of the parameter server architecture is not necessarily multiple machines